array.c

```diff
@@ -5270,15 +5270,18 @@ rb_ary_hash_values(long len, const VALUE *elements)
 
 /*
  *  call-seq:
- *    array.hash -> integer
+ *    hash -> integer
  *
  *  Returns the integer hash value for +self+.
  *
- *  Two arrays with the same content will have the same hash code (and will compare using #eql?):
+ *  Two arrays with the same content have the same hash value (and are #eql?):
```

Comment:
This is not true. Having the same hash value does not guarantee that they are `#eql?`. However, if they are `#eql?`, then they must have the same hash value.

The original documentation meant that `#hash` is used for the hash and `#eql?` is used for comparison in operations that hash.

Reply:
Fixed.